In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import (
    accuracy_score,
    classification_report,
    confusion_matrix
)

In [14]:
#Getting dataset in the colab
from google.colab import files

uploaded = files.upload()

In [15]:
#Verify if the correct files were uploaded
import os
os.listdir()

['.config',
 'dataset-tickets-german_normalized_50_5_2.csv',
 'aa_dataset-tickets-multi-lang-5-2-50-version.csv',
 'dataset-tickets-multi-lang3-4k.csv',
 '.ipynb_checkpoints',
 'dataset-tickets-multi-lang-4-20k.csv',
 'dataset-tickets-german_normalized.csv',
 'sample_data']

**DATABASE COMPARISION AND SELECTION**

In [16]:
datasets = {
    "multi-lang3-4k":
        "dataset-tickets-multi-lang3-4k.csv",

    "german-normalized-50":
        "dataset-tickets-german_normalized_50_5_2.csv",

    "german-normalized":
        "dataset-tickets-german_normalized.csv",

    "multi-lang-4-20k":
        "dataset-tickets-multi-lang-4-20k.csv",

    "multi-lang-5-2-50":
        "aa_dataset-tickets-multi-lang-5-2-50-version.csv"
}

In [17]:
comparison = []

for name, file_path in datasets.items():

    df_temp = pd.read_csv(file_path)

    comparison.append({
        "Dataset": name,
        "Rows": len(df_temp),
        "Columns": len(df_temp.columns),
        "Queues": df_temp["queue"].nunique(),
        "Languages": df_temp["language"].nunique(),
        "Missing Body": df_temp["body"].isnull().sum(),
        "Missing Queue": df_temp["queue"].isnull().sum(),
        "Duplicate Rows": df_temp.duplicated().sum(),
        "Smallest Queue": df_temp["queue"].value_counts().min()
    })

comparison_df = pd.DataFrame(comparison)

comparison_df

,Dataset,Rows,Columns,Queues,Languages,Missing Body,Missing Queue,Duplicate Rows,Smallest Queue
0,multi-lang3-4k,4000,17,10,5,1,0,0,53
1,german-normalized-50,13178,5,42,1,0,0,0,230
2,german-normalized,2125,5,10,1,0,0,0,170
3,multi-lang-4-20k,20000,15,10,2,2,0,0,263
4,multi-lang-5-2-50,28587,16,10,2,0,0,0,405


### Dataset Selection Rationale

Five candidate IT service desk datasets were evaluated to select the most appropriate dataset for the ticket classification and routing model. The datasets were compared based on the number of records, number of routing queues, language coverage, missing values, duplicate records, and the number of samples in the smallest routing class.

Based on this comparison, **`aa_dataset-tickets-multi-lang-5-2-50-version.csv`** was selected as the primary dataset. It contains **28,587 records across 10 routing queues and 2 languages**, making it the largest multilingual dataset among the candidates. It also contains **no missing values in the `body` or `queue` fields**, which are the primary input and target variables for the classification task, and no exact duplicate records. Additionally, its smallest routing queue contains **405 tickets**, providing better class representation than the other multilingual datasets.

Although the `multi-lang3-4k` dataset provides greater language diversity with **5 languages**, it contains only **4,000 records**, and its smallest routing class has only **53 tickets**. Therefore, some language diversity was traded for a substantially larger training set and stronger class representation. Overall, the selected dataset provides a good balance of **data volume, multilingual representation, data completeness, and class coverage**, making it suitable for developing and evaluating the initial ticket classification models.
